src/plugins/intel_cpu/src/nodes/matmul.cpp

```diff
@@ -110,8 +110,75 @@ bool MatMul::isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op,
     return true;
 }
 
+class MMShapeInfer : public ShapeInferEmptyPads {
+public:
+    MMShapeInfer(const std::shared_ptr<const ngraph::opset1::MatMul>& op) : m_matmul(op) {
+        m_out_rank = m_matmul->get_output_partial_shape(0).rank().get_length();
+        m_YShape = VectorDims(m_out_rank, 1); // for output and cache
+    }
+    std::vector<VectorDims> infer(
+        const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
+        const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
+        VectorDims AShape = input_shapes[0].get();
+        VectorDims BShape = input_shapes[1].get();
+        const size_t& Aranks = AShape.size();
+        const size_t& Branks = BShape.size();
```

Comment:
Here in opposite. The size of the `ptr` is the same as the `size_t` so there is no sense to take by reference.
```suggestion
        const size_t rankA = AShape.size();
        const size_t rankB = BShape.size();
```
Regarding the naming. Usually we try to start variables and method names with a lowercase letter.

Reply:
> Here in opposite. The size of the `ptr` is the same as the `size_t` so there is no sense to take by reference.
> 
> Regarding the naming. Usually we try to start variables and method names with a lowercase letter.

Done